### DISCUSSION TITLE
Configure the Kubernetes Ingresses in the Endpoint section

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#239] [Ideas]
### Which area of this project could be most improved?

Console

### Description

Hi guys,

I find the Endpoint section of the Mia-Platform Console very useful to handle in one place paths, security and documentation of my endpoints.

However, as for today, this section forces you to use Mia-Platform propretary API Gateway (either Nginx or Envoy).

It would be great if I could use this section, with less features (e.g. only path rewrite), for configuring Kubernetes Ingresses.
The advantages could be twofold:
- allow to use the Endpoints section without adding in the runtime any Mia-Platform propetary software
- allow to configure Kubernetes Ingresses from Mia-Platform Console

This isssue is somehow related to [this one](https://github.com/mia-platform/community/discussions/160)

What do you guys think about it?

Thanks & Regards.

**Comment by davidebianchi** [comment]
Hi @Lucascanna! Envoy is not of Mia-Platform, it's possible to use the open source solution. Also for the Mia-Platform API Gateway, it could be possible to use nginx plain with a small configuration and the generated configuration should work. 
One of the improvement we want to add to the section is to configure an nginx plain in fact. And it is possible also to add all the API Gateway, it is only needed a translation of the Console endpoint configuration.

Anyhow, as said in #160, we want to add ingress configuration from the Console, maybe using some CRD like the Gateway API. In this way, it will be possible to use the ingress without other services and manage all the configuration through the Console. And also expose some host (with cert-manager integration also with tls)
So, I agree with you to add this feature and we will add it in the roadmap.

**Reply by lucascanna** [reply]
Sorry @davidebianchi, my fault.
I was convinved that the Envoy plugin in the Mia-Platform Marketplace was a propetary wrapper of the Open Source solution (as for the Nginx API Gateway). 

Btw, as you also mentioned, very often the ingress features are enough for our use cases and the API Gateway adds a useless hop in the request chain. I am glad to hear you agree with me on this.

I look forward to see this feature in future versions.
Thanks.
